mpc-core/src/protocols/plain.rs

```diff
@@ -193,12 +241,172 @@ impl<F: PrimeField> PrimeFieldMpcProtocol<F> for PlainDriver<F> {
         dst[dst_offset..dst_offset + len].clone_from_slice(&src[src_offset..src_offset + len]);
     }
 
+    fn print_share(&self, to_print: &Self::FieldShare) {
+        if to_print.is_zero() {
+            println!("0")
+        } else {
+            println!("{to_print}")
+        }
+    }
+
     fn print(&self, to_print: &Self::FieldShareVec) {
-        print!("[");
         for a in to_print.iter() {
-            print!("{a}, ")
+            if a.is_zero() {
+                println!("0")
+            } else {
+                println!("{a}")
+            }
         }
-        println!("]");
+    }
+
+    fn index_sharevec(sharevec: &Self::FieldShareVec, index: usize) -> Self::FieldShare {
+        sharevec[index]
+    }
+
+    fn set_index_sharevec(sharevec: &mut Self::FieldShareVec, val: Self::FieldShare, index: usize) {
+        sharevec[index] = val;
+    }
+
+    fn sharevec_len(sharevec: &Self::FieldShareVec) -> usize {
+        sharevec.len()
```

Comment:
This should also probably be a trait that is added to fieldsharevec

Reply:
added a trait